app/controllers/posts_controller.rb

```diff
@@ -2,14 +2,7 @@
 
 class PostsController < ApplicationController
   def index
-    # top_post_id = Post.joins(:votes).group(:post_id).count.sort.to_h
-    top_post_id = Post.joins(:votes).group(:post_id, :id).select('posts.id, count(post_id) as count').order('count asc').last
-    @top_post = if !top_post_id.nil?
-                  Post.find(top_post_id.id)
-                else
-                  Post.last
-                end
-    @posts = Post.where.not(id: top_post_id)
+    @top_post, @posts = ListPosts.run!
```

Comment:
To akurat jest dobry materiał na ViewModel, nie na interaktor.
Interaktory dostarczają narzędzia do obsługi logiki biznesowej:
- definiowanie przyjmowanych parametrów i ich typów
- validacja danych
a w tym przypadku tego nie potrzebujemy.

Przykład view modelu: https://github.com/piotrimbierowicz/test_app_uwr/blob/master/app/view_models/home_page_view_model.rb